cmd/dep/ensure_test.go

```diff
@@ -186,3 +186,102 @@ func TestDeduceConstraint(t *testing.T) {
 		}
 	}
 }
+
+func TestEnsureUpdate(t *testing.T) {
+	test.NeedsExternalNetwork(t)
+	test.NeedsGit(t)
+
+	h := test.NewHelper(t)
+	defer h.Cleanup()
+
+	h.TempDir("src")
+	h.Setenv("GOPATH", h.Path("."))
+
+	m := `package main
+
+import (
```

Comment:
Note that only the unit tests have the golden-file pattern right now.  Integration test conversions are waiting on the CLI checks to pass `dep remove`.